### ISSUE TITLE
Integration with tools like rbenv or chruby

### ISSUE THREAD

**Issue by jacobbednarz** [Shopify/shadowenv#71]
Interested to hear about how Shopify is integrating Ruby into shadowenv and swapping versions between directories. Right now, most projects come with a `.ruby-version` file but I can't see a sane way of pulling the value from that file and either setting an environment variable (for tools like rbenv) or shimming in particular versions. I see the documentation moves some values in the environment but it doesn't really keep the source of truth as the `.ruby-version` file. Would something like `(open ".ruby-version")` somewhere in the shadowenv configuration cover this?

**Comment by burke** [comment]
We wouldn't want to do add extra capabilities to shadowlisp itself, but we have considered adding 'generators' or something: a simple command to run to generate useful shadowlisp into the .shadowenv.d directory from (say) a `.ruby-version` file.

This would be something along the lines of `shadowenv import .ruby-version` or `shadowenv generate ruby .ruby-version`, or something like that.

**Comment by jacobbednarz** [comment]
Ok, that leads me to my next question, what are you using for ruby version management with shadowenv and how does that interaction work? I can't imagine every application is operating on the same ruby (or node, or java) version.

**Comment by lugray** [comment]
We use our internal command line too `dev` to manage dependencies for out projects, so it's `dev`'s job to install the ruby if needed and write out the shadowlisp when users first run `dev up` for a project. You can see this project's `dev` configuration in the `dev.yml` in the root. For an example for a ruby project, you can see one from [cli-kit](https://github.com/Shopify/cli-kit/blob/master/dev.yml):
```yml
up:
  - ruby: 2.7.0
  - bundler

commands:
  test: bin/testunit
  style: "bundle exec rubocop -D"
```

which generates shadowlisp:
`.shadowenv.d/550_dev_ruby.lisp`
```lisp
(provide "ruby" "2.7.0")

(when-let ((ruby-root (env/get "RUBY_ROOT")))
 (env/remove-from-pathlist "PATH" (path-concat ruby-root "bin"))
 (when-let ((gem-root (env/get "GEM_ROOT")))
   (env/remove-from-pathlist "PATH" (path-concat gem-root "bin")))
 (when-let ((gem-home (env/get "GEM_HOME")))
   (env/remove-from-pathlist "PATH" (path-concat gem-home "bin"))))

(env/set "GEM_PATH" ())
(env/set "GEM_HOME" ())
(env/set "RUBYOPT" ())

(env/set "RUBY_ROOT" "/opt/rubies/2.7.0")
(env/prepend-to-pathlist "PATH" "/opt/rubies/2.7.0/bin")
(env/set "RUBY_ENGINE" "ruby")
(env/set "RUBY_VERSION" "2.7.0")
(env/set "GEM_ROOT" "/opt/rubies/2.7.0/lib/ruby/gems/2.7.0")

(when-let ((gem-root (env/get "GEM_ROOT")))
  (env/prepend-to-pathlist "GEM_PATH" gem-root)
  (env/prepend-to-pathlist "PATH" (path-concat gem-root "bin")))

(let ((gem-home
      (path-concat (env/get "HOME") ".gem" (env/get "RUBY_ENGINE") (env/get "RUBY_VERSION"))))
  (do
    (env/set "GEM_HOME" gem-home)
    (env/prepend-to-pathlist "GEM_PATH" gem-home)
    (env/prepend-to-pathlist "PATH" (path-concat gem-home "bin"))))
```

**Comment by jacobbednarz** [comment]
Bugger, I was afraid of that 🙁 Good to know though! While it would be possible to try and keep an environment variable in sync with the `.ruby-version` file contents, I feel it would kind of defeat the purpose of having a single source of truth if we were constantly ensuring it was manually in sync.

Happy to close this off for now as I've got all the answers I was after but would be interested to get a ping if a `read-file` or `env/get-from-file` type of method was added to shadowenv making it usable outside of `dev` tooling.
